File: Week 2/w2_q3.py
n,target=map(int,input().split())
m=[int(x) for x in input().split()]
s={}
for i in range(n):
    for j in range(i+1,n):
        k=target-m[i]-m[j]
        if k in s and s[k]!=i+1 and s[k]!=j+1:
            print(f"{s[k]} {i+1} {j+1}")
            exit()

    s[m[i]]=i+1
           

print("IMPOSSIBLE")     
    

File: Week 2/w2_q3.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n;
    long long x;
    cin >> n >> x;
    vector<pair<long long, int>> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i].first;
        a[i].second = i + 1;
    }
    sort(a.begin(), a.end());
    for (int i = 0; i < n - 2; i++) {
        long long target = x - a[i].first;
        int left = i + 1;
        int right = n - 1;
        while (left < right) {
            long long current_sum = a[left].first + a[right].first;

            if (current_sum == target) {
                cout << a[i].second << " " << a[left].second << " " << a[right].second << "\n";
                return 0;
            } 
            else if (current_sum < target) {
                left++;
            } 
            else {
                right--;
            }
        }
    }
    cout << "IMPOSSIBLE\n";
}
